examples/plotting/zoom_map_example.py

```diff
@@ -0,0 +1,65 @@
+"""
+=========================
+Zoomed-in cutout of a Map
+=========================
+
+This example demonstrates how to showcase a particular region of a full disk
+image within the same figure.
+
+This is a useful for figures that want to provide an overview of your subject.
+"""
+
+import matplotlib.pyplot as plt
+from matplotlib.patches import ConnectionPatch
+
+import astropy.units as u
+from astropy.coordinates import SkyCoord
+
+import sunpy.map
+from sunpy.data.sample import AIA_171_IMAGE
+
+###############################################################################
+# We will start off by defining a region of interest that we want to focus on.
+# In this case, we want to focus on the center of the Sun.
+length = 200 * u.arcsec
+x0 = -50 * u.arcsec
+y0 = -30 * u.arcsec
+
+###############################################################################
+# Now we will create a Map and then crop the region of interest.
+smap = sunpy.map.Map(AIA_171_IMAGE)
+bottom_left = SkyCoord(x0 - length, y0-length,
+                       frame=smap.coordinate_frame)
+top_right = SkyCoord(x0 + length, y0+length,
+                     frame=smap.coordinate_frame)
+submap = smap.submap(bottom_left, top_right=top_right)
+
```

Comment:
I cannot remember if this is possible when constructing a plot in sphinx-gallery, but if possible, I think it would be good to break up the plotting code a bit and have the explanatory text show up as text rather than code comments

Reply:
Thank you so much @wtbarnes for reviews. Is this fulfilling! please let me know.

![zoomed](https://user-images.githubusercontent.com/72349558/109108170-01261600-7759-11eb-82fd-5b31b14b87ad.png)

> Currently, the cutout is not centered on any particular region of interest. This (approximately) centers the FOV on the active region

Will, how you figure this out.  I still didn't get a clear picture how cutout wasn't centered on any particular region of interest. Could you please elaborate a bit on this.

Reply:
> I cannot remember if this is possible when constructing a plot in sphinx-gallery, but if possible, I think it would be good to break up the plotting code a bit and have the explanatory text show up as text rather than code comments

Are we talking about something like this:
```
# You can separate code blocks using either a single line of ``#``'s
# (>=20 columns), ``#%%``, or ``# %%``. For consistency, it is recommend that
# you use only one of the above three 'block splitter' options in your project.
```

Reply:
You will need a header line as well: 
```
###############################################################################
```

Reply:
I think the axis labels of the left image should go.

Reply:
yeah! was thinking about this...i will look into how to do it!